WhisperAPI/WhisperAPI/Models/MegaGenial/DistanceCalculator.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+
+namespace WhisperAPI.Models.MegaGenial
+{
+    public class DistanceCalculator
+    {
+        public static int[] ConvertWord(string word)
+        {
+            return word.ToCharArray().Select(n => Convert.ToInt32(n)).ToArray();
+        }
+
+        public static int GetDistance(int[] word1, int[] word2)
+        {
+            return DamerauLevenshteinDistance(word1, word2, 2);
+        }
+
+        private static void Swap<T>(ref T arg1, ref T arg2)
+        {
+            T temp = arg1;
+            arg1 = arg2;
+            arg2 = temp;
+        }
+
+        private static int DamerauLevenshteinDistance(int[] source, int[] target, int threshold)
```

Comment:
I'm just going to trust you that this method makes sense XD

On pourrait utiliser un NuGet pour ça
https://www.nuget.org/packages?q=Levenshtein

Sinon peut-être rajouter une coupe de tests de DamerauLevenshteinDistance() pour nous empecher de briser l'algo par accident